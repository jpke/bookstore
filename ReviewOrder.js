import React from 'react';
import { StyleSheet,
         Text,
         TextInput,
         TouchableHighlight,
         View
       } from 'react-native';

export default class ReviewOrder extends React.Component {
  constructor(props) {
    super(props);
  }

  navigate(routeName) {
    if(routeName === "HomeScreen") {
      this.props.navigator.popToTop(0);
    } else {
      this.props.navigator.push({
        name: routeName
      });
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.selectedBook}>
          <Text style={styles.title}>Review Order</Text>
            <View>
              <Text>{this.props.bookToOrder.title}</Text>
            </View>
            <View>
              <Text>{this.props.creditCardInfo.nameOnCard || "incomplete"}</Text>
              <Text>{this.props.creditCardInfo.cardNumber || "incomplete"}</Text>
              <Text>{this.props.creditCardInfo.securityCode || "incomplete"}</Text>
            </View>
            <View>
              <Text>{this.props.useDeliveryAddressAsBilling ? "Billing and Delivery Address" : "Billing Address"}</Text>
              <Text>{this.props.billingAddress.lineOne || "incomplete"}</Text>
              {this.props.billingAddress.lineTwo && <Text>{this.props.billingAddress.lineTwo || "incomplete"}</Text>
              }
              <Text>{this.props.billingAddress.city || "incomplete"}</Text>
              <Text>{this.props.billingAddress.state || "incomplete"}</Text>
              <Text>{this.props.billingAddress.zip || "incomplete"}</Text>
            </View>
            {!this.props.useDeliveryAddressAsBilling &&
              <View>
                <Text>Delivery Address</Text>
                <Text>{this.props.deliveryAddress.lineOne || "incomplete"}</Text>
                {this.props.deliveryAddress.lineTwo && <Text>{this.props.deliveryAddress.lineTwo || "incomplete"}</Text>}
                <Text>{this.props.deliveryAddress.city || "incomplete"}</Text>
                <Text>{this.props.deliveryAddress.state || "incomplete"}</Text>
                <Text>{this.props.deliveryAddress.zip || "incomplete"}</Text>
              </View>
            }
        </View>
        <View style={styles.menu}>
          <TouchableHighlight style={styles.button} onPress={() => {
              this.props.submitInfo(this.props.bookToOrder, 'SubmitOrder');
              this.navigate('OrderHistory');}
          }>
            <Text style={styles.menuItem}>Submit Order</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.button} onPress={() => this.props.navigator.pop()}>
            <Text style={styles.menuItem}>Back</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.button} onPress={() => this.navigate('HomeScreen')}>
            <Text style={styles.menuItem}>Home</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  selectedBook: {
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  title: {
    fontSize: 25,
  },
  input: {
    height: 50,
    width: 250,
    fontSize: 18,
    borderWidth: 1,
    borderColor: '#6799FF',
    marginTop: 3,
    padding: 5,
    borderRadius: 5
  },
  bookAuthor: {
    fontSize: 15,
    padding: 5
  },
  menu: {
    height: 200,
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  menuItem: {
    fontSize: 18,
    color: '#FFFFFF',
  },
  button: {
    backgroundColor: '#6799FF',
    width: 250,
    padding: 10,
    margin: 5,
    borderRadius: 10,
    alignItems: 'center',
  }
});
