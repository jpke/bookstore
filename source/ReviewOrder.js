import React from 'react';
import {connect} from 'react-redux';
import {submitInfo} from './actions';
import { StyleSheet,
         ScrollView,
         Text,
         TextInput,
         TouchableHighlight,
         View
       } from 'react-native';

class ReviewOrder extends React.Component {
  constructor(props) {
    super(props);
  }

  navigate(routeName) {
    if(routeName === "HomeScreen") {
      this.props.navigator.popToTop(0);
    } else {
      this.props.navigator.push({
        name: routeName
      });
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.selectedBook}>
          <View style={styles.header}>
            <View style={{flex: 1}}>
              <TouchableHighlight style={styles.homeButton} onPress={() => this.navigate('HomeScreen')}>
                <Text style={styles.menuItem}>Home</Text>
              </TouchableHighlight>
            </View>
            <Text style={styles.title}>Review Order</Text>
            <View style={{flex: 1}}></View>
          </View>
          <View style={styles.orderInfo}>
            <ScrollView>
              <View style={styles.innerBookInfo}>
                <Text style={styles.bookTitle}>{this.props.bookToOrder.title}</Text>
                <Text style={styles.bookAuthor}>{this.props.bookToOrder.author}</Text>
              </View>
              <View style={styles.itemInfo}>
                <Text style={styles.itemHeader}>Payment Information</Text>
                <Text style={styles.infoTitle}>name on card:</Text>
                <Text style={styles.info}> { this.props.creditCardInfo.nameOnCard}</Text>
                <Text style={styles.infoTitle}>card #:</Text>
                <Text style={styles.info}>
                  {this.props.creditCardInfo.cardNumber}</Text>
                <Text style={styles.infoTitle}>security code:</Text>
                <Text style={styles.info}>
                  {this.props.creditCardInfo.securityCode}</Text>
              </View>
              <View style={styles.itemInfo}>
                <Text style={styles.itemHeader}> {this.props.useDeliveryAddressAsBilling ? "Billing and Delivery Address" : "Billing Address"}</Text>
                <Text style={styles.infoTitle}>line one:</Text>
                <Text style={styles.info}>
                  {this.props.billingAddress.lineOne}</Text>
                <Text>
                  {this.props.billingAddress.lineTwo &&
                    <Text style={styles.infoTitle}>
                      line two:
                    </Text>
                  }
                </Text>
                <Text style={styles.infoContainer}>
                  {this.props.billingAddress.lineTwo &&
                    <Text style={styles.info}>
                      {this.props.billingAddress.lineTwo}
                    </Text>
                  }
                </Text>
                <Text style={styles.infoTitle}>city:</Text>
                <Text style={styles.info}>
                  {this.props.billingAddress.city}</Text>
                <Text style={styles.infoTitle}>state:</Text>
                <Text style={styles.info}>
                  {this.props.billingAddress.state}</Text>
                <Text style={styles.infoTitle}>zip:</Text>
                <Text style={styles.info}>
                  {this.props.billingAddress.zip}</Text>
              </View>
              {!this.props.useDeliveryAddressAsBilling &&
                <View style={styles.itemInfo}>
                  <Text style={styles.itemHeader}>Delivery Address</Text>
                  <Text style={styles.infoTitle}>line one:</Text>
                  <Text style={styles.info}>
                    {this.props.deliveryAddress.lineOne}</Text>
                  <Text>
                    {this.props.deliveryAddress.lineTwo &&
                      <Text style={styles.infoTitle}>
                        line two:
                      </Text>
                    }
                  </Text>
                  <Text style={styles.infoContainer}>
                    {this.props.deliveryAddress.lineTwo &&
                      <Text style={styles.info}>
                        {this.props.deliveryAddress.lineTwo}
                      </Text>
                    }
                  </Text>
                  <Text style={styles.infoTitle}>city:</Text>
                  <Text style={styles.info}>
                    {this.props.deliveryAddress.city}</Text>
                  <Text style={styles.infoTitle}>state:</Text>
                  <Text style={styles.info}>
                    {this.props.deliveryAddress.state}</Text>
                  <Text style={styles.infoTitle}>zip:</Text>
                  <Text style={styles.info}>
                    {this.props.deliveryAddress.zip}</Text>
                </View>
              }
            </ScrollView>
          </View>
        </View>
        <View style={styles.menu}>
          <TouchableHighlight style={styles.button} onPress={() => this.props.navigator.pop()}>
            <Text style={styles.menuItem}>Back</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.button} onPress={() => {
              this.props.submitInfo(this.props.bookToOrder, 'SubmitOrder');
              this.navigate('OrderHistory');}
          }>
            <Text style={styles.menuItem}>Submit Order</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    userInfo: state.userInfo,
    bookToOrder: state.bookToOrder,
    creditCardInfo: state.creditCardInfo,
    useDeliveryAddressAsBilling: state.useDeliveryAddressAsBilling,
    billingAddress: state.billingAddress,
    deliveryAddress: state.deliveryAddress
  }
}

function mapDispatchToProps(dispatch) {
  return {
    submitInfo: (info, type, option) => dispatch(submitInfo(info, type, option))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ReviewOrder);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  header: {
    marginTop: 15,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'flex-start'
  },
  selectedBook: {
    flex: 6,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  title: {
    flex: 3,
    fontSize: 40,
    color: '#ffcd67',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4,
    textAlign: 'center'
  },
  orderInfo: {
    flex: 5,
    width: 320,
  },
  itemHeader: {
    fontSize: 26,
    color: '#fff',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4,
    textAlign: 'center'
  },
  itemInfo: {
    flex: 3,
    marginTop: 6,
    marginBottom: 6,
    padding: 6,
    backgroundColor: 'rgba(255,205,103,0.8)',
    borderRadius: 5,
  },
  innerBookInfo: {
    flex: 4,
    backgroundColor: '#ffcd67',
    borderRadius: 5,
    padding: 10,
    alignItems: 'center',
    marginBottom: 10,
  },
  bookTitle: {
    fontSize: 28,
    padding: 5,
    textAlign: 'center'
  },
  bookAuthor: {
    fontSize: 24,
    padding: 5,
    textAlign: 'center'
  },
  infoTitle: {
    fontSize: 24,
    color: 'black',
  },
  infoContainer: {
    marginLeft: 60
  },
  info: {
    fontSize: 24,
    color: '#fff',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4,
    padding: 5,
    marginLeft: 60,
  },
  menu: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  menuItem: {
    fontSize: 18,
    color: '#FFFFFF',
  },
  homeButton: {
    backgroundColor: '#6799FF',
    width: 80,
    margin: 5,
    paddingTop: 10,
    paddingBottom: 10,
    borderRadius: 10,
    alignItems: 'center',
  },
  button: {
    backgroundColor: '#6799FF',
    // flex: 1,
    width: 150,
    padding: 10,
    margin: 10,
    borderRadius: 10,
    alignItems: 'center',
  }
});
