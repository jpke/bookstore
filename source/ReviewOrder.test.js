import React from 'react';
import ReactNative from 'react-native';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from './store';
const store = configureStore();

import ReviewOrder from './ReviewOrder';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ReviewOrder />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
