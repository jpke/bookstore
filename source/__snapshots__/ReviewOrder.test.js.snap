// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 6,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#6799FF",
                "borderRadius": 10,
                "margin": 5,
                "paddingBottom": 10,
                "paddingTop": 10,
                "width": 80,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
              }
            }
          >
            Home
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffcd67",
            "flex": 3,
            "fontSize": 40,
            "textAlign": "center",
            "textShadowColor": "black",
            "textShadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "textShadowRadius": 4,
          }
        }
      >
        Review Order
      </Text>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 5,
          "width": 320,
        }
      }
    >
      <ScrollView>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffcd67",
              "borderRadius": 5,
              "flex": 4,
              "marginBottom": 10,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 28,
                "padding": 5,
                "textAlign": "center",
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 24,
                "padding": 5,
                "textAlign": "center",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255,205,103,0.8)",
              "borderRadius": 5,
              "flex": 3,
              "marginBottom": 6,
              "marginTop": 6,
              "padding": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 26,
                "textAlign": "center",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            Payment Information
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            name on card:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
             
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            card #:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            security code:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255,205,103,0.8)",
              "borderRadius": 5,
              "flex": 3,
              "marginBottom": 6,
              "marginTop": 6,
              "padding": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 26,
                "textAlign": "center",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
             
            Billing Address
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            line one:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginLeft": 60,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            city:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            state:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            zip:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255,205,103,0.8)",
              "borderRadius": 5,
              "flex": 3,
              "marginBottom": 6,
              "marginTop": 6,
              "padding": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 26,
                "textAlign": "center",
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            Delivery Address
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            line one:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginLeft": 60,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            city:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            state:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 24,
              }
            }
          >
            zip:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 60,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            
          </Text>
        </View>
      </ScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#6799FF",
            "borderRadius": 10,
            "margin": 10,
            "padding": 10,
            "width": 150,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Back
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#6799FF",
            "borderRadius": 10,
            "margin": 10,
            "padding": 10,
            "width": 150,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Submit Order
      </Text>
    </View>
  </View>
</View>
`;
